File: src/main/java/com/github/nsc/de/compiler/parser/node/variables/VariableUsageNode.java
package com.github.nsc.de.compiler.parser.node.variables;

import com.github.nsc.de.compiler.parser.node.IdentifierNode;
import com.github.nsc.de.compiler.parser.node.ValuedNode;

public class VariableUsageNode implements ValuedNode {

    private final IdentifierNode variable;

    public VariableUsageNode(IdentifierNode variable) {
        this.variable = variable;
    }
    
    public IdentifierNode getVariable() {
        return variable;
    }

    @Override
    public String toString() {
        return "{variable="  + this.variable + '}';
    }
}

File: shake/compiler/parser/src/commonMain/kotlin/io/github/shakelang/shake/parser/node/variables/VariableUsageNode.kt
package io.github.shakelang.shake.parser.node.variables

import io.github.shakelang.shake.parser.node.ValuedNode
import io.github.shakelang.shake.parser.node.IdentifierNode
import io.github.shakelang.parseutils.characters.position.PositionMap

class VariableUsageNode(map: PositionMap, val variable: IdentifierNode) : ValuedNode(map) {
    override fun toJson(): Map<String, *> = mapOf("name" to "VariableUsageNode", "variable" to variable.json)
}